near-sdk/src/types/account_id.rs

```diff
@@ -0,0 +1,174 @@
+use borsh::{BorshDeserialize, BorshSchema, BorshSerialize};
+use serde::{de, Deserialize, Serialize};
+use std::convert::TryFrom;
+use std::fmt;
+
+use crate::env::is_valid_account_id;
+
+/// Account identifier. Provides access to user's state.
+#[derive(
+    Debug,
+    Clone,
+    PartialEq,
+    PartialOrd,
+    Ord,
+    Eq,
+    BorshDeserialize,
+    BorshSerialize,
+    Serialize,
+    Hash,
+    BorshSchema,
+)]
+pub struct AccountId(String);
+
+impl AccountId {
+    /// Returns reference to the account ID bytes.
+    pub fn as_bytes(&self) -> &[u8] {
+        self.0.as_bytes()
+    }
+    /// Returns reference to the account ID string.
+    pub fn as_str(&self) -> &str {
+        self.0.as_str()
+    }
+    /// Consumes `self` to return inner [`String`] representation.
+    pub fn into_string(self) -> String {
+        self.0
+    }
+    // TODO this should probably be marked as unstable or crate scoped
```

Comment:
how do we want to handle this, with the unstable feature flag we've discussed at times?

Reply:
Yeah I'd like to get others input about this. The conversion before was also not so certain (related #312) so it'd be nice to have others input to make sure we make a good long term API decision